app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -3946,6 +3952,31 @@ class BrowserTabViewModel @Inject constructor(
         newTabPixels.get().fireNewTabDisplayed()
     }
 
+    private suspend fun shouldPromptForBrokenSiteReport(): Boolean {
+        return withContext(dispatchers.io()) {
+            val refreshCount = brokenSitePrompt.getUserRefreshesCount()
+            when (refreshCount) {
+                0, 1 -> false
+                2 -> {
+                    blockListPixelsPlugin.get2XRefresh()?.getPixelDefinitions()?.forEach {
+                        pixel.fire(it.pixelName, it.params)
+                    }
+                    pixel.fire(AppPixelName.RELOAD_TWICE_WITHIN_12_SECONDS)
+                    Timber.d("KateTest--> send pixel only bc 2 refreshes")
+                    false
+                }
+                else -> {
+                    blockListPixelsPlugin.get3XRefresh()?.getPixelDefinitions()?.forEach {
+                        pixel.fire(it.pixelName, it.params)
+                    }
+                    pixel.fire(AppPixelName.RELOAD_THREE_TIMES_WITHIN_20_SECONDS)
+                    Timber.d("KateTest--> should show bc 3 refreshes")
+                    site?.let { brokenSitePrompt.shouldShowBrokenSitePrompt(it.url) } ?: false
```

Comment:
I guess null site here is an edge case here, so probably fine if pixel and prompt fall out of sync